src/internal/functions/Remove-AzOpsDeployment.ps1

```diff
@@ -21,11 +25,145 @@
+                if ($depPolicySetDefinition) {
+                    $depPolicySetDefinition = foreach ($policySetDefinition in $depPolicySetDefinition) {
+                        $policy = Get-AzPolicySetDefinition -Id $policySetDefinition.Id -ErrorAction SilentlyContinue
+                        if ($policy) {
+                            $dependency += [PSCustomObject]@{
+                                ResourceType = $policy.ResourceType
+                                ResourceId = $policy.PolicySetDefinitionId
+                            }
+                            $dependency += Get-AzPolicyAssignmentDeletionDependency -resourceToDelete $policy
+                        }
+                    }
+                }
+                if ($dependency) {
+                    $dependency = $dependency | Sort-Object ResourceId -Unique | Where-Object {$_.ResourceId -ne $resourceToDelete.ResourceId}
+                    return $dependency
+                }
+            }
+        }
+        function Search-AzGraphDeletionDependency {
+            param (
+                $query,
+
+                $PartialMgDiscoveryRoot = (Get-PSFConfigValue -FullName 'AzOps.Core.PartialMgDiscoveryRoot')
+            )
+            $results = @()
+            if($PartialMgDiscoveryRoot) {
+                foreach ($managementRoot in $PartialMgDiscoveryRoot) {
+                    $processing = Search-AzGraph -Query $query -ManagementGroup $managementRoot
+                    if ($processing) {
+                        $results += $processing
+                        DO {
+                            if ($processing.SkipToken) {
+                                $processing = Search-AzGraph -Query $query -ManagementGroup $managementRoot -SkipToken $processing.SkipToken
+                                $results += $processing
+                            }
+                            else {
+                                $done = $true
+                            }
+                        } While ($done -ne $true)
+                    }
+                }
+            }
+            else {
+                $processing = Search-AzGraph -Query $query -UseTenantScope
+                if ($processing) {
+                    $results += $processing
+                    DO {
+                        if ($processing.SkipToken) {
+                            $processing = Search-AzGraph -Query $query -UseTenantScope -SkipToken $processing.SkipToken
+                            $results += $processing
+                        }
+                        else {
+                            $done = $true
+                        }
+                    } While ($done -ne $true)
+                }
+            }
+            if ($results) {
+                $results = $results | Sort-Object Id -Unique
+                return $results
+            }
+        }
+        $script:dependencyMissing = $null
```

Comment:
Why do we need to use $script variable scope here?

Reply:
We did not 👯‍♂️